### DISCUSSION TITLE
[AppCheck][Android] SafetyNet Attestation API sunsets in June 2023

### DISCUSSION THREAD

**Discussion by atanaskanchev** [invertase/react-native-firebase#6586] [Ideas] [answered]
<div align="center" class="emphasis" style="font-style: normal; font-variant-caps: normal; font-weight: 400; letter-spacing: normal; orphans: auto; text-indent: 0px; text-transform: none; white-space: normal; widows: auto; word-spacing: 0px; -webkit-text-size-adjust: auto; -webkit-text-stroke-width: 0px; text-decoration: none; caret-color: rgb(0, 0, 0); color: rgb(0, 0, 0); font-family: Helvetica; padding: 20px 0px; font-size: 18px; line-height: 26px;"><em>Review the deprecation phases and migrate your use of the Google SafetyNet Attestation API to avoid any disruptions in detecting risky interactions with your app.</em></div><p style="font-style: normal; font-variant-caps: normal; font-weight: 400; letter-spacing: normal; orphans: auto; text-align: start; text-indent: 0px; text-transform: none; white-space: normal; widows: auto; word-spacing: 0px; -webkit-text-size-adjust: auto; -webkit-text-stroke-width: 0px; text-decoration: none; caret-color: rgb(0, 0, 0); color: rgb(0, 0, 0); font-family: Helvetica; font-size: 12px;">Hello Android Developer,</p><p style="font-style: normal; font-variant-caps: normal; font-weight: 400; letter-spacing: normal; orphans: auto; text-align: start; text-indent: 0px; text-transform: none; white-space: normal; widows: auto; word-spacing: 0px; -webkit-text-size-adjust: auto; -webkit-text-stroke-width: 0px; text-decoration: none; caret-color: rgb(0, 0, 0); color: rgb(0, 0, 0); font-family: Helvetica; font-size: 12px;">We’re writing to let you know that, as announced on<span class="Apple-converted-space"> </span><strong>May 11, 2022,</strong><span class="Apple-converted-space"> </span>the<span class="Apple-converted-space"> </span><a href="https://notifications.google.com/g/p/AKWoLQhY4AX-e3ABfIQA1gp6y6tNRCy9LRgdxQJt3_XbGEZKqWQbEqRodPvF6vPezme9Kw0F-8nm8pnsEow04KWf0P7WudLpcBeaxS1lKrzWM3CFiwN8PnShyVKwX0SIj2W44qA" target="_blank"><strong>Play Integrity API</strong></a><span class="Apple-converted-space"> </span>will consolidate several device integrity offerings (including the SafetyNet Attestation integrity verdict) under a single API.</p><p style="font-style: normal; font-variant-caps: normal; font-weight: 400; letter-spacing: normal; orphans: auto; text-align: start; text-indent: 0px; text-transform: none; white-space: normal; widows: auto; word-spacing: 0px; -webkit-text-size-adjust: auto; -webkit-text-stroke-width: 0px; text-decoration: none; caret-color: rgb(0, 0, 0); color: rgb(0, 0, 0); font-family: Helvetica; font-size: 12px;">Starting<span class="Apple-converted-space"> </span><strong>June 2023</strong>, we’ll start to gradually phase out the<span class="Apple-converted-space"> </span><a href="https://notifications.google.com/g/p/AKWoLQjCGNnWlfjyBrnxhpgvXcSGdF_Nr69BZ4DIpxavQ-uqwZHEo2Ic_1kGr1l3soC0ZaxJnqrRYquj6jzeeAwZXFhNoOao-CmoRnh9L5ecfnokGSpQ1cR3Gphe3k2O0SsOnIuKVl2YRH4Peic" target="_blank">SafetyNet Attestation API</a>. This slow shift will give you time to migrate to the new<span class="Apple-converted-space"> </span><strong>Play Integrity API</strong><span class="Apple-converted-space"> </span>and avoid disruptions for your business.</p><h3 style="font-style: normal; font-variant-caps: normal; letter-spacing: normal; orphans: auto; text-align: start; text-indent: 0px; text-transform: none; white-space: normal; widows: auto; word-spacing: 0px; -webkit-text-size-adjust: auto; -webkit-text-stroke-width: 0px; text-decoration: none; caret-color: rgb(0, 0, 0); color: rgb(0, 0, 0); font-family: Helvetica;">What do I need to know?</h3><p style="font-style: normal; font-variant-caps: normal; font-weight: 400; letter-spacing: normal; orphans: auto; text-align: start; text-indent: 0px; text-transform: none; white-space: normal; widows: auto; word-spacing: 0px; -webkit-text-size-adjust: auto; -webkit-text-stroke-width: 0px; text-decoration: none; caret-color: rgb(0, 0, 0); color: rgb(0, 0, 0); font-family: Helvetica; font-size: 12px;">The<span class="Apple-converted-space"> </span><strong>Play Integrity API</strong><span class="Apple-converted-space"> </span>includes all the integrity signals that SafetyNet Attestation offers and more, such as Google Play user licenses and better error messaging. The new API is designed so you can easily take advantage of future features with minimal build time required to upgrade.</p>

Deprecation Phase | Phase Description
-- | --
MigrationJune 2022 | You should start integrating with the Play Integrity API as soon as you can. In the meantime, SafetyNet Attestation will continue to work for your app.After November 2022, we’ll add a new field to the SafetyNet Attestation API attestation response with information regarding the transition.We’ll publish more information soon on the public mailing list and in the public documentation.
TransitionJune 30, 2023 | If you’ve migrated to the Play Integrity API, SafetyNet Attestation will continue to work on earlier versions of your app. This way you can still detect risky interactions in earlier versions of your app.If you have not migrated, SafetyNet Attestation won’t work in any version of your app and will return an error.SafetyNet Attestation will detect that you have migrated when your app calls the Play Integrity API in production.
Turn offJune 30, 2024 | The SafetyNet Attestation API will no longer work in any version of your app. Your app will receive an error.

<h3 style="font-style: normal; font-variant-caps: normal; letter-spacing: normal; orphans: auto; text-align: start; text-indent: 0px; text-transform: none; white-space: normal; widows: auto; word-spacing: 0px; -webkit-text-size-adjust: auto; -webkit-text-stroke-width: 0px; text-decoration: none; caret-color: rgb(0, 0, 0); color: rgb(0, 0, 0); font-family: Helvetica;">What do I need to do?</h3><p style="font-style: normal; font-variant-caps: normal; font-weight: 400; letter-spacing: normal; orphans: auto; text-align: start; text-indent: 0px; text-transform: none; white-space: normal; widows: auto; word-spacing: 0px; -webkit-text-size-adjust: auto; -webkit-text-stroke-width: 0px; text-decoration: none; caret-color: rgb(0, 0, 0); color: rgb(0, 0, 0); font-family: Helvetica; font-size: 12px;">We recommend you<span class="Apple-converted-space"> </span><strong>migrate to the Play Integrity API</strong><span class="Apple-converted-space"> </span>immediately to avoid any disruptions in detecting risky interactions with your app and fighting abuse.</p><ul style="font-style: normal; font-variant-caps: normal; font-weight: 400; letter-spacing: normal; orphans: auto; text-align: start; text-indent: 0px; text-transform: none; white-space: normal; widows: auto; word-spacing: 0px; -webkit-text-size-adjust: auto; -webkit-text-stroke-width: 0px; text-decoration: none; caret-color: rgb(0, 0, 0); color: rgb(0, 0, 0); font-family: Helvetica; font-size: 12px;"><li>If you’ve defined your anti-abuse strategy and are using the SafetyNet Attestation API, you’ll need to<span class="Apple-converted-space"> </span><strong><a href="https://notifications.google.com/g/p/AKWoLQgppflKOAelfVubjp68TczKJ6hibC5LWBB1RKKQsd77s5PrRCqtDQIW0MIR7-MezL_0MkNhzIyzASkbmyK0PWS-LYFBUh9-xqNIpHWof8RHVdV76hc0DGJTxLN1y1BXxKSCXtfObbY" target="_blank">set up</a><span class="Apple-converted-space"> </span>the Play Integrity API</strong><span class="Apple-converted-space"> </span>from your<span class="Apple-converted-space"> </span><a href="https://notifications.google.com/g/p/AKWoLQhBy6YMEaSCDm_44NEbz2n0kOfPwIQ5GbatOalpbeljmOD92di3e8M4FkUJvboZfkGauCiF4JPdDm6BPhdqOmo7DcX4aoz-8ds6cnZtoTjS1fwEMnpGwEXSjrFw" target="_blank">Play Console</a>, then<span class="Apple-converted-space"> </span><strong>follow the<span class="Apple-converted-space"> </span><a href="https://notifications.google.com/g/p/AKWoLQgWqNZt0D8-P9d_zexIv0FWr8TCHE09qqgukcuQiwUtURYG7c4FyC4dhRlxOHs7uRmGzeuJwEvc0DRorty_70YOZB4h_5bGcd9MQR41rVDsss3gTtGtTXvP0F6GHHPjkKBURBqkaQM-mw" target="_blank">migration guide</a></strong>.</li><li>If you’re not distributing your app through Google Play, you can<span class="Apple-converted-space"> </span><strong>use the Play Integrity API from the Google Cloud Console by following these<span class="Apple-converted-space"> </span><a href="https://notifications.google.com/g/p/AKWoLQj4_K8e4N4EipSe0YG2kGZ2ED0AUOvvlJwAj35dWGVz1ESiBB_K8KMX9F1591mnygbZiq7a2c3P5keTOqs3F7VOcJa2B-yXXMBAvQ-V4HycOhJkEwtXB3LND0lOjnDLj8rodqrEONQZmN1sWxhGQ3Xcrm4LX48-kEHgbScMl_M4OV2zU89QMqAVut62LeztlZEAVoBdysX7" target="_blank">steps</a></strong>. By using this process, you don't have to create a Play developer account or publish a release in the Google Play Console.</li></ul><br class="Apple-interchange-newline" style="caret-color: rgb(0, 0, 0); color: rgb(0, 0, 0); font-style: normal; font-variant-caps: normal; font-weight: 400; letter-spacing: normal; orphans: auto; text-align: start; text-indent: 0px; text-transform: none; white-space: normal; widows: auto; word-spacing: 0px; -webkit-text-size-adjust: auto; -webkit-text-stroke-width: 0px; text-decoration: none;"><br class="Apple-interchange-newline">Review the deprecation phases and migrate your use of the Google SafetyNet Attestation API to avoid any disruptions in detecting risky interactions with your app.
Hello Android Developer,

We’re writing to let you know that, as announced on May 11, 2022, the [Play Integrity API](https://notifications.google.com/g/p/AKWoLQhY4AX-e3ABfIQA1gp6y6tNRCy9LRgdxQJt3_XbGEZKqWQbEqRodPvF6vPezme9Kw0F-8nm8pnsEow04KWf0P7WudLpcBeaxS1lKrzWM3CFiwN8PnShyVKwX0SIj2W44qA) will consolidate several device integrity offerings (including the SafetyNet Attestation integrity verdict) under a single API.

Starting June 2023, we’ll start to gradually phase out the [SafetyNet Attestation API](https://notifications.google.com/g/p/AKWoLQjCGNnWlfjyBrnxhpgvXcSGdF_Nr69BZ4DIpxavQ-uqwZHEo2Ic_1kGr1l3soC0ZaxJnqrRYquj6jzeeAwZXFhNoOao-CmoRnh9L5ecfnokGSpQ1cR3Gphe3k2O0SsOnIuKVl2YRH4Peic). This slow shift will give you time to migrate to the new Play Integrity API and avoid disruptions for your business.

What do I need to know?

The Play Integrity API includes all the integrity signals that SafetyNet Attestation offers and more, such as Google Play user licenses and better error messaging. The new API is designed so you can easily take advantage of future features with minimal build time required to upgrade.

Deprecation Phase	Phase Description
Migration
June 2022	
You should start integrating with the Play Integrity API as soon as you can. In the meantime, SafetyNet Attestation will continue to work for your app.
After November 2022, we’ll add a new field to the SafetyNet Attestation API attestation response with information regarding the transition.
We’ll publish more information soon on the [public mailing list](https://notifications.google.com/g/p/AKWoLQjAuc5N2V_ntl-nPKJmvHYLKbAvBGGTKLxFyr4r-qmIekv_8lVF4qD20h77ttoTvXl0kXHPtGcioEXR4paaJHMIdwD4mt7RzbOZ6inuPaG0yrqMOjdE0jItJcjaxmOG) and in the [public documentation](https://notifications.google.com/g/p/AKWoLQjCGNnWlfjyBrnxhpgvXcSGdF_Nr69BZ4DIpxavQ-uqwZHEo2Ic_1kGr1l3soC0ZaxJnqrRYquj6jzeeAwZXFhNoOao-CmoRnh9L5ecfnokGSpQ1cR3Gphe3k2O0SsOnIuKVl2YRH4Peic).
Transition
June 30, 2023	
If you’ve migrated to the Play Integrity API, SafetyNet Attestation will continue to work on earlier versions of your app. This way you can still detect risky interactions in earlier versions of your app.
If you have not migrated, SafetyNet Attestation won’t work in any version of your app and will return an error.
SafetyNet Attestation will detect that you have migrated when your app calls the Play Integrity API in production.
Turn off
June 30, 2024	The SafetyNet Attestation API will no longer work in any version of your app. Your app will receive an error.
What do I need to do?

We recommend you migrate to the Play Integrity API immediately to avoid any disruptions in detecting risky interactions with your app and fighting abuse.

If you’ve defined your anti-abuse strategy and are using the SafetyNet Attestation API, you’ll need to [set up](https://notifications.google.com/g/p/AKWoLQgppflKOAelfVubjp68TczKJ6hibC5LWBB1RKKQsd77s5PrRCqtDQIW0MIR7-MezL_0MkNhzIyzASkbmyK0PWS-LYFBUh9-xqNIpHWof8RHVdV76hc0DGJTxLN1y1BXxKSCXtfObbY) the Play Integrity API from your [Play Console](https://notifications.google.com/g/p/AKWoLQhBy6YMEaSCDm_44NEbz2n0kOfPwIQ5GbatOalpbeljmOD92di3e8M4FkUJvboZfkGauCiF4JPdDm6BPhdqOmo7DcX4aoz-8ds6cnZtoTjS1fwEMnpGwEXSjrFw), then follow the [migration guide](https://notifications.google.com/g/p/AKWoLQgWqNZt0D8-P9d_zexIv0FWr8TCHE09qqgukcuQiwUtURYG7c4FyC4dhRlxOHs7uRmGzeuJwEvc0DRorty_70YOZB4h_5bGcd9MQR41rVDsss3gTtGtTXvP0F6GHHPjkKBURBqkaQM-mw).
If you’re not distributing your app through Google Play, you can use the Play Integrity API from the Google Cloud Console by following these [steps](https://notifications.google.com/g/p/AKWoLQj4_K8e4N4EipSe0YG2kGZ2ED0AUOvvlJwAj35dWGVz1ESiBB_K8KMX9F1591mnygbZiq7a2c3P5keTOqs3F7VOcJa2B-yXXMBAvQ-V4HycOhJkEwtXB3LND0lOjnDLj8rodqrEONQZmN1sWxhGQ3Xcrm4LX48-kEHgbScMl_M4OV2zU89QMqAVut62LeztlZEAVoBdysX7). By using this process, you don't have to create a Play developer account or publish a release in the Google Play Console.

**Comment by mikehardy** [comment, marked as answer]
Search: #6434
